formats/src/main/java/io/parsingdata/metal/format/UID.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2013-2016 Netherlands Forensic Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.parsingdata.metal.format;
+
+import static io.parsingdata.metal.Shorthand.cat;
+import static io.parsingdata.metal.Shorthand.con;
+
+import java.math.BigInteger;
+import java.nio.ByteBuffer;
+import java.util.UUID;
+
+import io.parsingdata.metal.encoding.Encoding;
+import io.parsingdata.metal.expression.value.ConstantFactory;
+import io.parsingdata.metal.expression.value.ValueExpression;
+
+/**
+ * Constants for Microsoft GUID and Java UUID.
+ *
+ * @author Netherlands Forensic Institute.
+ */
+public class UID {
+    private static final Encoding ENC = new Encoding();
+
+    /**
+     * Use a String representation of a GUID as predicate.
+     * {@code eq(guid("caa16737-fa36-4d43-b3b6-33f0aa44e76b"))}
+     * @param guid GUID, for example "caa16737-fa36-4d43-b3b6-33f0aa44e76b"
+     * @return expression to use as predicate
+     */
+    public static ValueExpression guid(final String guid) {
+        final String[] parts = guid.split("-");
+        if (parts.length != 5) {
+            throw new IllegalArgumentException("Invalid GUID string: " + guid);
+        }
+
+        // Note that GUID bytes differ from UUID bytes, as the first 3 parts are reversed
+        // Use ByteBuffer instead of long to make sure no leading zeroes are omitted
+        final ByteBuffer buffer = ByteBuffer.allocate(16);
+        buffer.putInt(0, Integer.reverseBytes((int) Long.parseLong(parts[0], 16)));
```

Comment:
The endianness of the first three are defined as "Native". This means that we need to be able to define the encoding of those fields when we create a `guid` value expression.

See see *Binary encoding* section in the [Wikipedia GUID article](https://en.wikipedia.org/wiki/Globally_unique_identifier).